kernel/byterun/coq_interp.c

```diff
@@ -1413,9 +1413,10 @@ value coq_interprete
         int b;
         Uint63_eq0(b, *sp);
         if (b) {
+                  value t = accu;
           AllocPair();
-          Field(accu, 0) = *sp;
-          Field(accu, 1) = *sp++;
+          Field(accu, 0) = *sp++;
+          Field(accu, 1) = t;
```

Comment:
That seems wrong. If the `AllocPair` allocation fails and the garbage collector runs, then the value pointed by `accu` will get relocated, thus causing `t` to become dangling. You could use the stack to avoid the issue, but perhaps the more readable solution is to not use `AllocPair` here, so as to keep `accu` around. Something like:
```c
value block;
Alloc_small(block, 2, coq_tag_pair);
Field(block, 0) = accu;
Field(block, 1) = *sp++;
accu = block;
```

Reply:
I see (in your code probably the values of the fields are reversed).

Reply:
Right, `accu` is the numerator, so it should go to the second component.